articles/2019-07-09-responsive-sans-media-queries.md

```diff
@@ -0,0 +1,294 @@
+    </article>
+  </div>
+
+  <div style="flex: 1; padding: 0 10px;">
+    <article style="background: #bfb;">
+      <h2 style="margin: 0; padding: 20px; font: 900 64px monospace;">Green</h2>
+    </article>
+  </div>
+
+  <div style="flex: 1; padding: 0 10px;">
+    <article style="background: #bbf;">
+      <h2 style="margin: 0; padding: 20px; font: 900 64px monospace;">Blue</h2>
+    </article>
+  </div>
+</section>
+```
+
+Ce changement n’a aucun impact sur le rendu grand format, mais cela va nous
+permettre en petit format d’obtenir le rendu suivant :
+
+<iframe
+    allowtransparency="true" allowfullscreen="true" scrolling="no" frameborder="no"
+    height="300" style="width: 100%; max-width: 300px;"
+    title="Responsive without MQs, step 3"
+    src="//codepen.io/MoOx/embed/NZBvLB/?height=300&theme-id=light&default-tab=result"  >
+</iframe>
+
+En fonction du contenu intérieur des blocs que vous allez avoir, vous allez
+pouvoir vous utilisez plutôt `min-width` ou `flex-basis`. Je vous laisse jouer
+un peu avec histoire de vous faire la main.
+
+> [Codepen un peu plus lisible](https://codepen.io/MoOx/pen/bPjyVm?editors=1100)
+
+En fait je n’ai que cette astuce.
+
+Je plaisante à peine. Car si on ajoute à cela le côté malin de
+`overflow: hidden` pour cacher de l'information optionel, on peut faire des
+trucs assez puissant.
+
+Regardons ça avec un exemple plus complexe: on va imaginer le header d'un site.
+
+En appliquant cette technique à l’extrême, (ce qui n’est pas une quantité de
+travail astronomique, et reste quelque chose de propre et tout à fait
+maintenable, surtout avec une approche composant et non document) on peut se
+retrouver avec un code très simple, sans MQs qui donnerait les rendus suivant:
+
+<iframe
+    allowtransparency="true" allowfullscreen="true" scrolling="no" frameborder="no"
+    height="300" style="width: 1000px; margin: auto;"
+    title="Responsive without MQs, real world example"
+    src="//codepen.io/MoOx/embed/WqKBGm/?height=300&theme-id=light&default-tab=result"  >
+</iframe>
+
+<iframe
+    allowtransparency="true" allowfullscreen="true" scrolling="no" frameborder="no"
+    height="300" style="width: 100%; max-width: 300px;"
+    title="Responsive without MQs, real world example"
+    src="//codepen.io/MoOx/embed/WqKBGm/?height=300&theme-id=light&default-tab=result"  >
+</iframe>
+
+```html
+<todo />
+```
+
+> [Code de cette example sur Codepen](https://codepen.io/MoOx/pen/WqKBGm)
+
+Des exemples de ce type là, surtout en exploitant bien flex-basis, peuvent se
+révéler extrêmement puissant. On peut très bien se contenter ça. Après comme
+souvent lorsqu’on avoir des besoins plus complexe, il sera à vous de juger si
+continuer à utiliser cette technique est pertinent, ou s’il est plus judicieux
+d’utiliser des MQs afin d’éviter de vous défoncer le cerveau. Ou alors de faire
+du rendu conditionnel avec `window.matchMedia` si votre platforme vous le
```

Comment:
`matchMedia` still not relevant?

Reply:
Yeah this was stupid. I could put "Ou alors de faire du rendu conditionnel avec JavaScript"?